pilot/pkg/config/kube/gateway/route_collections.go

```diff
@@ -771,12 +771,43 @@ func mergeHTTPRoutes(baseVirtualServices krt.Collection[RouteWithKey], opts ...k
 		sortRoutesByCreationTime(configs)
 		base := configs[0].DeepCopy()
 		baseVS := base.Spec.(*istio.VirtualService)
-		for _, config := range configs[1:] {
+		for i, config := range configs[1:] {
 			thisVS := config.Spec.(*istio.VirtualService)
 			baseVS.Http = append(baseVS.Http, thisVS.Http...)
 			// append parents
 			base.Annotations[constants.InternalParentNames] = fmt.Sprintf("%s,%s",
 				base.Annotations[constants.InternalParentNames], config.Annotations[constants.InternalParentNames])
+			// Merge Extra field (especially for InferencePool configs)
+			if config.Extra != nil {
+				if base.Extra == nil {
```

Comment:
Can we pull this out a level and do a check like this?
```suggestion
				if base.Extra == nil && config.Extra != nil { base.Extra = make(map[string]any) }
```

We would still need the `if config.Extra != nil` check above, but I think it would be easier to read with even one less level of nested ifs

Reply:
Done!, I applied the change to simplify